firefox-ios/Client/Frontend/Browser/Toolbars/Redux/AddressBarState.swift

```diff
@@ -555,4 +556,21 @@ struct AddressBarState: StateType, Equatable {
             a11yLabel: .TabToolbarForwardAccessibilityLabel,
             a11yId: AccessibilityIdentifiers.Toolbar.forwardButton)
     }
+
+    private static func getAddressToolbarBorderPosition(
+        action: ToolbarAction,
+        state: AddressBarState)
+    -> AddressToolbarBorderPosition? {
+        guard let toolbarState = store.state.screenState(ToolbarState.self, for: .toolbar, window: action.windowUUID)
+        else { return state.borderPosition }
+
+        let isWebPage = action.url?.isWebPage() ?? false
```

Comment:
Figma says that the navigation border should also be shown when the toolbar is set to top and we are on the homepage. 
Also I am not sure why the middleware is not used to get the correct position anymore. I don't really like the mix between having logic here and also using Redux as the source of truth. So far the middleware was handling the toolbar border position using the `ToolbarManager`. Is there a reason we don't use that anymore but also leave the code in place?

Reply:
<img width="414" alt="Screenshot 2024-09-23 at 5 16 30 PM" src="https://github.com/user-attachments/assets/68dfff94-9f15-4d4a-ad24-db6df064ba33">

Maybe I am misunderstanding the spec, but to me, it reads that it should only be visible on scroll when the toolbar is set to the top and on the homepage, but not when the page is scroll to the top.

I put this logic directly in the reducer instead of the middleware because it had no dependencies (if you don't include `ToolbarManager`). This code is specifically for when the url changes and we either navigate to a webpage and show the border, or navigate to the homepage and hide the border. We still use the middleware (and `ToolbarManager`) for monitoring for homepage scroll. I will take a look and see if there is a more consistent way to write this where the logic isn't as spread out, because I agree, it is confusing